pkg/autoscaler/aggregation/bucketing.go

```diff
@@ -88,10 +96,21 @@ func NewTimedFloat64Buckets(window, granularity time.Duration) *TimedFloat64Buck
 	// e.g. 60s / 2s = 30.
 	nb := math.Ceil(float64(window) / float64(granularity))
 	return &TimedFloat64Buckets{
-		buckets:         make([]float64, int(nb)),
-		granularity:     granularity,
-		window:          window,
-		decayMultiplier: computeDecayMultiplier(nb),
+		buckets:     make([]float64, int(nb)),
+		granularity: granularity,
+		window:      window,
+	}
+}
+
+// NewWeightedFloat64Buckets generates a new WeightedFloat64Buckets with the given
+// granularity.
+func NewWeightedFloat64Buckets(window, granularity time.Duration) *WeightedFloat64Buckets {
+	// Number of buckets is `window` divided by `granularity`, rounded up.
+	// e.g. 60s / 2s = 30.
+	nb := math.Ceil(float64(window) / float64(granularity))
+	return &WeightedFloat64Buckets{
+		TimedFloat64Buckets: NewTimedFloat64Buckets(window, granularity),
+		smoothingCoeff:     computeDecayMultiplier(nb),
```

Comment:
Ooh Coefficient- I think that's exactly what I was thinking of but couldn't actual think of!

Reply:
Should change computeDecayMultiplier to match tho

Reply:
Done.